src/v/cluster/cluster_utils.cc

```diff
@@ -332,7 +333,8 @@ partition_raft_state get_partition_raft_state(consensus_ptr ptr) {
             state.last_flushed_log_index = md.last_flushed_log_index;
             state.match_index = md.match_index;
             state.next_index = md.next_index;
-            state.last_sent_offset = md.last_sent_offset;
+            state.last_sent_offset = md.in_flight_offset.value_or(
```

Comment:
makes sense to rename here too IMO, as this API is used only for internal debug jsons